src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2313,6 +2336,33 @@ RunDistributedExecution(DistributedExecution *execution)
 }
 
 
+/*
+ * HasIncompleteConnectionEstablishment returns true if any of the connections
+ * that has been initiated by the executor is in initilization stage.
+ */
+static bool
+WaitForIncompleteConnectionEstablishment(DistributedExecution *execution)
```

Comment:
This method doesn't seem to be `waiting`, what about `HasIncompleteConnection`? Does it miss anything?